### PR TITLE
Update dashboard version to v1.8.3

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it:** Updates dashboard addon to the latest version. Changelog can be found [here](https://github.com/kubernetes/dashboard/releases/tag/v1.8.3).

**Release note**:

```release-note
Update dashboard version to v1.8.3
```

### CODE REVIEW

**Comment by floreks** [comment]
Is there a way to check logs of a container that is failing? Without that it is hard to debug.

**Comment by joelsmith** [comment]
@floreks Would it be possible to update this PR to go straight to a 1.8.3 as per https://github.com/kubernetes/dashboard/issues/2813 ?
Also, I have opened up a related issue, #59354 and I think that your PR here is probably the better place to fix it instead of my PR #59355 updating to 1.8.3.

File: cluster/addons/dashboard/dashboard-rbac.yaml [hunk 1]
```diff
@@ -7,10 +7,6 @@ metadata:
   name: kubernetes-dashboard-minimal
   namespace: kube-system
 rules:
-  # Allow Dashboard to create 'kubernetes-dashboard-key-holder' secret.
```
Comment (liggitt) [inline cluster/addons/dashboard/dashboard-rbac.yaml]: +1 for this change, probably need to add an empty EnsureExists `kubernetes-dashboard-key-holder` Secret to the secrets.yaml file

File: cluster/addons/dashboard/dashboard-controller.yaml [hunk 2]
```diff
@@ -30,7 +30,7 @@ spec:
       priorityClassName: system-cluster-critical
       containers:
       - name: kubernetes-dashboard
-        image: k8s.gcr.io/kubernetes-dashboard-amd64:v1.8.0
+        image: k8s.gcr.io/kubernetes-dashboard-amd64:v1.8.3
```
Comment (liggitt) [inline cluster/addons/dashboard/dashboard-controller.yaml]: can you also change the kubernetes-dashboard-certs volume back to a secret (reverting the dashboard change in 
https://github.com/kubernetes/kubernetes/pull/58720/files#diff-7ab88bb7654d97946d6328f11f29d177, now that the dashboard only reads from that mount)

**Comment by floreks** [comment]
Done.

**Comment by floreks** [comment]
PTAL

**Comment by liggitt** [comment]
needs squashing
/assign @mikedanese

**Comment by floreks** [comment]
@mikedanese rebased and squashed the commits.

**Comment by joelsmith** [comment]
/hold cancel
Removing hold since this PR is squashed now. The failing `pull-kubernetes-e2e-gke` test is currently not marked required and is failing across the vast majority of PRs. So now this PR just needs a lgtm. Sorry that I don't have enough background in this area to provide a review.

**Comment by floreks** [comment]
@liggitt could you take a look?

**Pull request merged**